docs/4.x/tracking-javascript-guide.md

```diff
@@ -765,22 +765,31 @@ If you want to ignore download or outlink tracking on a specific link, you can a
 
 ## Asking for consent
 
-In the context of the [GDPR privacy regulations](https://matomo.org/docs/gdpr/), when you are processing personal data, in some cases you will need to ask for your users' consent. To identify whether you need to ask for consent, you need to determine whether your lawful basis for processing personal data is "Consent" or "Legitimate interest", or whether you can avoid collecting personal data altogether. We recommend to learn more about the [lawful basis under the GDPR for Matomo](https://matomo.org/blog/2018/04/lawful-basis-for-processing-personal-data-under-gdpr-with-matomo/).
+In the context of the [GDPR privacy regulations](https://matomo.org/docs/gdpr/), when you are processing personal data, in some cases you will need to ask for your users' consent. To identify whether you need to ask for any consent, you need to determine whether your lawful basis for processing personal data is "Consent" or "Legitimate interest", or whether you can avoid collecting personal data altogether. We recommend to learn more about the [lawful basis under the GDPR for Matomo](https://matomo.org/blog/2018/04/lawful-basis-for-processing-personal-data-under-gdpr-with-matomo/).
+
+Matomo differentiates between cookie and tracking consent:
+* In the context of tracking consent no cookies will be used and no tracking request will be sent unless consent was given. As soon as consent was given, tracking requests will be sent and cookies will be used. 
+* In the context of cookie consent tracking requests will be always sent. However, cookies will be only used if consent for storing and using cookies was given by the user. [Learn how cookies impact reports accuracy](https://matomo.org/faq/general/faq_156/). This requires Matomo 3.14.0
            
-Follow the steps below to ask your user for their consents before their data is processed within Matomo.
+Follow the steps below to ask your user for either tracking or cookie consent before their data is processed within Matomo.
 
 ### Step 1: require consent
 
 To require consent, insert the following line at top of your existing Matomo Tracking code on all your pages:
 
 ```js
-// require user consent before processing data
+// require user tracking consent before processing data
 _paq.push(['requireConsent']);
+
+// OR require user cookie consent before storing and using any cookies
+_paq.push(['requireCookieConsent']);
+
 _paq.push(['trackPageView']);
 [...]
 ```
 
-Once this function `requireConsent` is executed then no tracking request will be sent to Matomo. Note that some [tracking cookies](https://matomo.org/faq/general/faq_146/) will be set during this page view but will be deleted as soon as the user navigates to a different page or closes the browser.
```

Comment:
the note is not true anymore since no cookies will be set anymore. We fixed this